prow/config/config_test.go

```diff
@@ -761,3 +761,58 @@ periodics:
 		}
 	}
 }
+
+// Integration test for fake secrets loading in a secret agent.
+// Checking also if the agent changes the secret's values as expected.
+func TestSecreAgentLoading(t *testing.T) {
+	tempTokenValue := "121f3cb3e7f70feeb35f9204f5a988d7292c7ba1"
+	changedTokenValue := "121f3cb3e7f70feeb35f9204f5a988d7292c7ba0"
+
+	// Creating a temporary directory.
+	secretDir, err := ioutil.TempDir("", "secretDir")
+	if err != nil {
+		t.Fatalf("fail to create a temporary directory: %v", err)
+	}
+	defer os.RemoveAll(secretDir)
+
+	// Create the first temporary secret.
+	firstTempSecret := filepath.Join(secretDir, "firstTempSecret")
+	if err := ioutil.WriteFile(firstTempSecret, []byte(tempTokenValue), 0666); err != nil {
+		t.Fatalf("fail to write secret: %v", err)
+	}
+
+	// Create the second temporary secret.
+	secondTempSecret := filepath.Join(secretDir, "secondTempSecret")
+	if err := ioutil.WriteFile(secondTempSecret, []byte(tempTokenValue), 0666); err != nil {
+		t.Fatalf("fail to write secret: %v", err)
+	}
+
+	// Starting the agent and add the two temporary secrets.
+	secretAgent := NewSecretAgent()
+	if err := secretAgent.Start([]string{firstTempSecret, secondTempSecret}); err != nil {
+		t.Fatalf("Error starting secrets agent. %v", err)
+	}
+
+	// Get the secrets from the agent.
+	secretsMap := secretAgent.Config()
+
+	// Check if the values are as expected.
+	for k, v := range secretsMap {
+		if string(v) != tempTokenValue {
+			t.Fatalf("In secret %s it was expected %s but found %s", k, tempTokenValue, string(v))
+		}
+	}
+
+	// Change the values
```

Comment:
Don't you want to change the files?

Reply:
Yes actually, I want to test how the agent is interacting to changes of the files.